samcli/local/docker/container.py

```diff
@@ -379,7 +380,26 @@ def copy(self, from_container_path, to_host_path):
             fp.seek(0)
 
             with tarfile.open(fileobj=fp, mode="r") as tar:
-                tar.extractall(path=to_host_path)
+
+                def is_within_directory(directory, target):
```

Comment:
it is better to move these inner functions to a utility module, and to add unit test cases for them.